storage/storage_dest.go

```diff
@@ -599,35 +638,42 @@ func (s *storageImageDestination) commitLayer(index int, info addedLayerInfo, si
 
 	// Check if there's already a layer with the ID that we'd give to the result of applying
 	// this layer blob to its parent, if it has one, or the blob's hex value otherwise.
-	// The diffIDOrTOCDigest refers either to the DiffID or the digest of the TOC.
-	diffIDOrTOCDigest, haveDiffIDOrTOCDigest := s.getDiffIDOrTOCDigest(info.digest)
-	if !haveDiffIDOrTOCDigest {
+	// The layerID refers either to the DiffID or the digest of the TOC.
+	layerID, haveLayerID := s.getLayerID(info.digest, info.tocDigest)
```

Comment:
Right now, `info` doesn’t tell whether we pulled by TOC, or by full blob.

Reply:
if `info.tocDigest != ""` then the layer was pulled by TOC